Node/call-gemini/functions/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ const {onCall, HttpsError} = require("firebase-functions/v2/https");
+ const {logger} = require("firebase-functions/v2");
+ const { GoogleGenerativeAI } = require("@google/generative-ai");
+ const { defineSecret } = require('firebase-functions/params');
+
+ const geminiToken = defineSecret("API_TOKEN", {
+  description: "Gemini API token. Created using " +
+      "https://ai.google.dev/tutorials/get_started_node#set-up-project",
+});
+
+ exports.callGemini = onCall({secrets: [geminiToken]}, async (request) => {
+
```

Comment:
Before calling Gemini, let's look for both appcheck and auth, but gate each with its own boolean parameter (https://firebase.google.com/docs/functions/config-env?gen=2nd#parameter_types) so devs can choose whether they want auth/app check or not.

Something like:

```js
// ...imports

const appCheckRequired = defineBoolean("APP_CHECK_REQUIRED");
const authRequired = defineBoolean("AUTH_REQUIRED");

// ...other setup

exports.callGemini = onCall(
  {
    enforceAppCheck: appCheckRequired, // Reject requests with missing or invalid App Check tokens.
  },
  (request) => {
    if (authRequired.val() && !request.auth) {
      throw new HttpsError(
        "failed-precondition",
        "The function must be called while authenticated.",
      );
    }

    // ...call Gemini
  },
);
```